packages/@tinacms/graphql/src/database/index.ts

```diff
@@ -649,9 +651,21 @@ export class Database {
             : [];
         }
 
+        let folderOps: BatchOp[];
+        if (collection) {
+          folderOps = makeFolderOpsForCollection(
+            folderTreeBuilder.tree,
+            collection,
+            collectionIndexDefinitions,
+            'put',
+            level
+          );
+        }
```

Comment:
without this folders were not getting created when specifying the a folder in the name of the file